pilot/adapter/config/file/config.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"io/ioutil"
+	"istio.io/istio/pilot/model"
+	"fmt"
+)
+
+const (
+	defaultNamespace = "default"
+	defaultDomain    = "cluster.local"
+)
+
+// Information for a single element of configuration stored in a file.
+type ConfigRef interface {
```

Comment:
Why the indirection here? I don't really think this warrants an interface, just make `configRef` public. Further, given the recommended API change below (to accept a `*model.ConfigMeta` rather than a set of strings) you don't even need the `NewConfigRef*` methods at all. Just let the user construct the objects themselves.

Reply:
done.